app/service/ApplicationService.scala

```diff
@@ -101,18 +101,49 @@ class ApplicationService @Inject()(connectorWrapper: ConnectorsWrapper,
     } yield apiVersions
   }
 
-  def subscribeToApi(id: String, context: String, version: String)(implicit hc: HeaderCarrier): Future[ApplicationUpdateSuccessful] =
-    connectorWrapper.forApplication(id).flatMap(_.thirdPartyApplicationConnector.subscribeToApi(id, APIIdentifier(context, version)))
+  def subscribeToApi(application: Application, context: String, version: String)(implicit hc: HeaderCarrier): Future[ApplicationUpdateSuccessful] = {
+    val connectors = connectorWrapper.connectorsForEnvironment(application.deployedTo)
```

Comment:
If the val 'connectors' isn't used anywhere then it should be removed.

Reply:
It's used on line 126 so i'll keep it in there.

Though the code could be inlined to make it less noisy :thinking:

Reply:
Ah I take it back then :)